src/dlsym_default.c

```diff
@@ -335,6 +299,99 @@ if (default_symbol_index) {
       }
     }
   }
+  *tags_p = tags;
+  *default_symbol_index_p = default_symbol_index;
+  
+  if (default_symbol_index)
+    return tags.base_addr + tags.symtab[default_symbol_index].st_value;
+  else
+    return NULL;
+}
+
+// Given a pseudo-handle, symbol name, and addr, returns the address of the symbol
+// with the given name of a default version found by the search order of the given
+// handle which is either RTLD_DEFAULT or RTLD_NEXT.
+void *dlsym_default_internal_flag_handler(void* handle, const char* symbol, void* addr,
+      dt_tag *tags_p, Elf32_Word *default_symbol_index_p)
+{
+  Dl_info info;
+  void* extra_info;
```

Comment:
I think this extra variable is not required. We could move the `struct link_map *map` here and use it directly in the call to `dladdr1()`.